modules/nextflow/src/main/groovy/nextflow/executor/FluxExecutor.groovy

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2022, Seqera Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package nextflow.executor
+import java.nio.file.Path
+import java.util.regex.Pattern
+
+import groovy.util.logging.Slf4j
+import nextflow.processor.TaskRun
+/**
+ * Processor for Flux Framework executor
+ *
+ * See https://flux-framework.org
+ *
+ *
+ * @author Paolo Di Tommaso <paolo.ditommaso@gmail.com>
+ * @author Vanessa Sochat <sochat1@llnl.gov>
+ */
+@Slf4j
+class FluxExecutor extends AbstractGridExecutor {
+
+    static final private Pattern SUBMIT_REGEX = ~/(ƒ.+)/
+
+    /**
+     * Gets the directives to submit the specified task to the cluster for execution
+     *
+     * @param task A {@link TaskRun} to be submitted
+     * @param result The {@link List} instance to which add the job directives
+     * @return A {@link List} containing all directive tokens and values.
+     */
+    protected List<String> getDirectives(TaskRun task, List<String> result) {
+        return result
+    }
+
+    // Flux does not require a special token or header
+    String getHeaderToken() { null }
+
+    /**
+     * The command line to submit this job
+     *
+     * @param task The {@link TaskRun} instance to submit for execution to the cluster
+     * @param scriptFile The file containing the job launcher script
+     * @return A list representing the submit command line
+     */
+    @Override
+    List<String> getSubmitCommandLine(TaskRun task, Path scriptFile ) {
+
+        List<String> result = ['flux', 'mini', 'submit']
+        result << '--setattr=cwd=' + quote(task.workDir)
+        result << '--job-name="' + getJobNameFor(task) + '"'
+        result << '--output=' + quote(task.workDir.resolve(TaskRun.CMD_LOG))  // -o OUTFILE
+
+        if( task.config.cpus > 1 ) {
+            result << '--cores=' + task.config.cpus.toString()
+        }
```

Comment:
I don't know Nextflow, but if `task.config.cpus` here means to allocate > 1 core per task, then I think you might want `--cores-per-task=` here instead of just `--cores`. I _think_ `--cores=N` here will end up running multiple tasks since the default `--tasks-per-core` is 1. (And sorry for the confusion there!)

Reply:
Yes, nextflow is cpus per task